src/sources/source.ts

```diff
@@ -2,13 +2,23 @@
  * Contract of all Sources.
  * Every source must implements the fetch mehod
  * in order to provide the avatar source.
- * 
+ *
  * @export
  * @interface Source
  */
 export interface Source {
     sourceType: string;
     sourceId: string;
+    readonly isAsync: boolean;
     getAvatar(size?: number): string;
 }
 
+/**
+ * Contract of all async sources.
+ * Every async source must implement the processResponse method that extracts the avatar url from the data
+ * @interface AsyncSource
+ */
+export interface AsyncSource extends Source {
```

Comment:
I see this as an abstract class which has on abstract method processResponse. Thus,
Async avatar classes must extend the AsyncSource class and implement the abstract method.
Hence, we will be able to check wether the source class is instanceOf AsyncSource rather
than using a boolean flag.
what do you think ?